test/test_rospy/test/rostest/test_basic_services.py

```diff
@@ -213,6 +214,28 @@ def test_String_String(self):
             resp_req_kwds = self._test_req_kwds(name, Cls, {'str': String('FOO'), 'str2': Val('bar')})
             for resp in [resp_req, resp_req_naked, resp_req_kwds]:
                 self.assertEquals('FOObar', resp.str.data)
+
+    def test_String_String_unicode(self):
+        from std_msgs.msg import String
+        from test_rospy.srv import StringString, StringStringRequest
+        from test_rospy.msg import Val
+        import sys
+        Cls = StringString
+        Req = StringStringRequest
+
+        for name in [STRING_CAT_SERVICE_NAKED, STRING_CAT_SERVICE_WRAPPED]:
+            if sys.version < '3':
```

Comment:
Right, but look at what happens in the two cases:

```
if sys.version_info.major < '3':
    # Python 2, returns a <unicode>.
    resp_req = self._test(name, Cls, Req(String(u'ロボット'), Val(u'机器人')))
else:
    # Python 3, returns a <str>.
    resp_req = self._test(name, Cls, Req(String('ロボット'), Val('机器人')))
```

But you can use the `u` prefix in both, so the following can replace the above:

```
# Returns <unicode> in Python 2, <str> in Python 3.
resp_req = self._test(name, Cls, Req(String(u'ロボット'), Val(u'机器人')))
```